Add notes for CG-08-04.md (#616)
WebAssembly/meetings: main/2020/CG-08-04.md

<unanimous consent>

BS: we will allow this.

##### See https://github.com/WebAssembly/spec/pull/1230#issuecomment-665669922

#### Poll: Should we update the spec to use LEB128 encoding for prefixed opcodes? [10 min]

##### The CG had a poll in May 2017 and decided to require prefix opcodes to be followed by a LEB128 opcode. However, the spec does not correctly reflect this, and some implementations differ in their behavior too.

BS: spec document does not say that the opcode is leb128 encoded, but we had a CG poll in May 2017 for leb 128 encoding. Since we decided this many years ago, i think it should be a spec mistake. But since there are multiple different implementations, we might want to discuss this.

JP: who are the divergences?

BS: binaryen wabt prefix + leb, v8 does prefix + byte, SM does prefix + leb.

TL: no visible difference right now?

BS: nope, long encoding

AR: extension operators and bulk operators

BS: bulk is not upstream yet, non-trapping. You can write those as a long encoding, that will diverge implementation.

AR: right now it is conservative, we won’t break anything if we allow LEB.

TL: i think we should make the change

BS: agree, any opposition?

> AR: this is something we screwed over other proposals

Reviewer: ```suggestion
AR: this is something we screwed up in proposals
```

AR: we could interpret this differently, for now we allow single byte LEB, but if we need more codespace, then 2 byte leb.

BS: danger there is that implementation have to care about that. Currently implementations already have to read LEB, it will be nicer to have uniformity.

KM: right now SIMD already have more, you have to support LEB if you have to support SIMD

BS: if you have a generic mechanism for opcode, if you read a prefix byte, you always LEB.

> AR: must be a LEB but no redundant bytes.

Reviewer: ```suggestion
AR: rephrase: must be a LEB but no redundant bytes.
```
